packages/blockchain-wallet-v4-frontend/src/data/balances/selectors.ts

```diff
@@ -359,19 +342,24 @@ export const getTotalWalletBalancesSorted = createDeepEqualSelector(
 export const getCoinNonCustodialBalance = (
   coin: CoinType
 ): ((state: RootState) => RemoteDataType<string, number | BigNumber>) => {
-  switch (true) {
-    case coin === 'BTC':
-      return __getBtcNonCustodialBalance
-    case coin === 'BCH':
-      return __getBchNonCustodialBalance
-    case coin === 'ETH':
-      return __getEthNonCustodialBalance
-    case coin === 'XLM':
-      return __getXlmNonCustodialBalance
-    case !!window.coins[coin].coinfig.type.erc20Address:
-      return __getErc20NonCustodialBalance(coin)
-    default:
-      throw Error(`Coin "${coin}" not supported by getCoinNonCustodialBalance`)
+  return (state: RootState) => {
+    return createDeepEqualSelector(
+      [selectors.core.data.coins.getUnifiedBalances],
+      (unifiedBalancesR) => {
+        let balance = new BigNumber(0)
+        const transform = (unifiedBalances: ExtractSuccess<typeof unifiedBalancesR>) => {
+          unifiedBalances
+            .filter(({ ticker }) => ticker === coin)
+            .forEach(({ amount }) => {
+              balance = balance.plus(amount ? amount.amount : 0)
+            })
+
+          return balance
+        }
+
+        return lift(transform)(unifiedBalancesR)
+      }
```

Comment:
Let's try to avoid redundant `lift` and `forEach` where we can.

```suggestion
    (unifiedBalancesR) =>
        unifiedBalancesR.map((unifiedBalances) =>
          unifiedBalances
            .filter(({ ticker }) => ticker === coin)
            .reduce(
              (balance, { amount }) => balance.plus(amount ? amount.amount : 0),
              new BigNumber(0)
            )
        )
```